src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SummarizeActiveRegionOutAgainstVCF.java

```diff
@@ -0,0 +1,105 @@
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+import picard.cmdline.programgroups.VariantEvaluationProgramGroup;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+/**
+ * This tool exists to overcome a limitation of the HaplotypeCaller '--assembly-region-out' argument in that it doesn't
+ * track the variants themselves. For some debugging/assembly engine evaluation work it is helpful to annotate the overlapping
+ * regions based on some subsetted VCF output. THis tool allows for that to happen.
+ */
+@CommandLineProgramProperties(
+        summary = "Annotates an AssemblyRegionOutput file from the haplotype caller with a count of overlapping variants.",
+        oneLineSummary = "Annotate AssemblyRegionOutput",
+        programGroup = VariantEvaluationProgramGroup.class
+)
+
+@ExperimentalFeature
+public class SummarizeActiveRegionOutAgainstVCF extends FeatureWalker<TableFeature> {
+    private static final Logger logger = LogManager.getLogger(SummarizeActiveRegionOutAgainstVCF.class);
+
+    @Argument(fullName = "active-region-summary", doc = "table output of active regions from haplotype caller or mutect")
+    File input = null;
+
+    @Argument(fullName = "output", doc = "output tsv to which to write the summary")
+    String outputPath = null;
+    private SimpleXSVWriter outputTableWriter;
+
+    @Argument(fullName= "variants-to-count", doc="The set of alleles to force-call regardless of evidence", optional=true)
+    public FeatureInput<VariantContext> overlappingVariantInput;
+
+    @Override
+    public void onTraversalStart() {
+        super.onTraversalStart();
+        try {
+            this.outputTableWriter = new SimpleXSVWriter(IOUtils.getPath(outputPath), '\t');
+            outputTableWriter.setHeaderLine(Lists.newArrayList("active_region", "kmers_assembled", "haplotypes_found", "variants_overlapping"));
+        } catch (IOException e) {
+            throw new GATKException("failed to open output writer");
+        }
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        Object ret = super.onTraversalSuccess();
+        try {
+            outputTableWriter.close();
+        } catch (IOException outputException) {
+            throw new GATKException("Failed when trying to close the output table", outputException);
+        }
+        return ret;
+    }
+
+    @Override
+    protected boolean isAcceptableFeatureType(Class<? extends Feature> featureType) {
+        return featureType.equals(TableFeature.class);
+    }
+
+    @Override
+    public void apply(TableFeature feature, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        List<VariantContext> overlappingVariants = featureContext.getValues(overlappingVariantInput);
+
+        outputTableWriter.getNewLineBuilder()
+                .setColumn("active_region", feature.get("active_region"))
```

Comment:
Is there any way to write this more explicitly as appending the count of overlapping variants to the input feature?